File: one_way.py
def one_way(string1, string2):
	l1 = len(string1)
	l2 = len(string2)
	if abs(l1 - l2) > 1:
		return False
	has_operated = False
	if abs(l1 - l2) == 1:
		if l1 > l2:
			i = 0
			while i < l2:
				if string1[i] != string2[i]:
					if not has_operated:
						string1 = string1[:i] + string1[i+1:]
						has_operated = True
					else:
						return False
				else:
					i = i + 1
		else:
			i = 0
			while i < l1:
				if string1[i] != string2[i]:
					if not has_operated:
						string1 = string1[:i] + string2[i] + string1[i:]
						has_operated = True
					else:
						return False
				else:
					i = i + 1
	else:
		i = 0
		while i < l1:
			if string1[i] != string2[i]:
				if not has_operated:
					has_operated = True
				else:
					return False
			i = i + 1
	return True

def main():
	string1 = "palxs"
	string2 = "pale"
	iow = one_way(string1, string2)
	print(iow)

if __name__ == "__main__":
	main()

File: one_way.cpp
#include <iostream>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std; 
 
bool oneWay(string str1,string str2){

	int l1 = str1.length();
	int l2 = str2.length();

	if( abs(l1-l2)>1 ){
		return false;
	}

	int has_operator = 0;
	int i = 0;
	char* cstr1 = (char*) str1.c_str();
	char* cstr2 = (char*) str2.c_str();
	
	while( i <= min(l1,l2) ){
		if(cstr1[i] != cstr2[i]){
			if(has_operator==0){
				if (l1>l2){
					cout<< "first string: "<< str1 <<endl;
					str1 = str1.substr (0, i) + str1.substr(i+1);
					cout<< "new string: "<< str1 <<endl;
					cout<< "second string: "<< str2 <<endl;
				}
				else if (l1<l2){
					cout << "first string: " << str1 << endl;
					cout << "second string: " << str2 << endl;
					char letter = cstr2[i];
					str1 = str1.substr(0,i) + letter + str1.substr(i);
					cout<< "new string: " << str1 << endl;
				}
			has_operator = 1;
			}
			else{
				return false;
			}
		}
		else {
		i = i+1;
     	}
    }
    return true;
}

int main(){

	string string1 = "pale";
	string string2 = "pami";

	cout << oneWay(string1,string2) <<endl;

	return 0; 
}
